### PR TITLE
CKEDITOR.plugins.link.getSelectedLink does not fetch all links

### LINKED ISSUE #859: Bug in CKEDITOR.plugins.link.getSelectedLink
## Are you reporting a feature request or a bug?
Bug

## Provide detailed reproduction steps (if any)

1. run ckeditor from [cdn](http://cdn.ckeditor.com/4.7.3/full-all/samples/)
1. run source mode
1. insert `<p><a class="linkClass" href="https://www.google.pl"> some button text </a></p>`
1. go back to WYSIWYG mode.
1. double-click on `some button text`

### Expected result
the link dialog should show with proper href data.

### Actual result

the link dialog shows without `href` and `class` data.

## Other details
What I found out so far:
- getSelectedLink (editor, false) works
- on right-click selection.getSelectedElement() is null, but when I double-click it points to the correct "a"
- on right-click range.getCommonAncestor() points to the "some button text", on double-click range.getCommonAncestor() points to the "p" tag <<<< this causes the problem in my opinion.

* Browser: Chrome
* OS: Windows
* CKEditor version: 4.7.2

### PR DESCRIPTION
**Pull request by wojtekw92**
## What is the purpose of this pull request?

Bug fix

## Does your PR contain necessary tests?

All patches which change the editor code must include tests. You can always read more
on [PR testing](http://docs.ckeditor.com/#!/guide/dev_contributing_code-section-tests),
[how to set the testing environment](http://docs.ckeditor.com/#!/guide/dev_tests) and
[how to create tests](http://docs.ckeditor.com/#!/guide/dev_tests-section-creating-your-own-test)
in the official CKEditor documentation.

### This PR contains

- [X] Unit tests
- [X] Manual tests

## What changes did you make?
Fix double click on anchor with image and text.

closes #859

### CODE REVIEW

File: tests/plugins/link/manual/getselectedlink.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+@bender-tags: link, 4.8.0, 859, bug
+@bender-ui: collapsed
+@bender-ckeditor-plugins: link,toolbar,wysiwygarea,elementspath,contextmenu,image
+
+1. Right-click on the span ("some button text") and select "Edit Link".
+
+**Expected:**
+* Link should have class `someClass`.
```
Comment (Comandeer) [inline tests/plugins/link/manual/getselectedlink.md]: Class described here is not the same as the class used inside the code.

File: tests/plugins/link/manual/getselectedlink.html [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+			<a class="linkClass" href="https://www.google.pl">
+					<img src="http://lorempixel.com/10/20/"/>
+					some button text
+			</a>
+			some text
+	</p>
+</div>
+
+<script>
+	CKEDITOR.replace( 'editor' /*,{ allowedContent: true }*/);
```
Comment (Comandeer) [inline tests/plugins/link/manual/getselectedlink.html]: This commented code should be deleted.

File: tests/plugins/link/link.js [hunk 3]
```diff
@@ -546,6 +546,20 @@
...
+		'test edit link with selction': function() {
+			var bot = this.editorBot;
+
+			bot.setData( '<p><a class="linkClass" href="linkUrl"><img src="someUrl"/>some button text</a>some text</p>', function() {
+				var editable = bot.editor.editable();
+
+				bot.editor.getSelection().selectElement( editable.findOne( 'a' ) );
+
+				bot.dialog( 'link', function( dialog ) {
+					assert.areSame( dialog.getValueOf( 'info', 'url' ), 'linkUrl' );
```
Comment (Comandeer) [inline tests/plugins/link/link.js]: It would be nice to close the dialog after the test.

**Comment by wojtekw92** [comment]
I was not able to reproduce this issue with table. The table is now available in manual test. @Comandeer please check this again to confirm.

**Pull request merged**